pkg/util/log/format_crdb_v1_test.go

```diff
@@ -0,0 +1,177 @@
+)
+
+func TestCrdbV1FormatDecode(t *testing.T) {
+	var templateEntry logpb.Entry
+	// We only use the datadriven framework for the ability to rewrite the output.
+	datadriven.RunTest(t, "testdata/crdb_v1", func(t *testing.T, td *datadriven.TestData) string {
+		switch td.Cmd {
+		case "common":
+			templateEntry = logpb.Entry{}
+			if err := json.Unmarshal([]byte(td.Input), &templateEntry); err != nil {
+				td.Fatalf(t, "invalid input syntax: %v", err)
+			}
+
+		case "entries":
+			var inputEntries []logpb.Entry
+			for _, line := range strings.Split(td.Input, "\n") {
+				entry := templateEntry
+				if err := json.Unmarshal([]byte(line), &entry); err != nil {
+					td.Fatalf(t, "invalid input syntax: %v", err)
+				}
+				inputEntries = append(inputEntries, entry)
+			}
+
+			var buf bytes.Buffer
+			// Encode.
+			for _, entry := range inputEntries {
+				_ = FormatLegacyEntry(entry, &buf)
+			}
+			// Decode.
+			entryStr := buf.String()
+			decoder := NewEntryDecoder(strings.NewReader(entryStr), WithMarkedSensitiveData)
+			var outputEntries []logpb.Entry
+			for {
+				var entry logpb.Entry
+				if err := decoder.Decode(&entry); err != nil {
+					if err == io.EOF {
+						break
+					}
+					td.Fatalf(t, "error while decoding: %v", err)
+				}
+				outputEntries = append(outputEntries, entry)
+			}
+			if len(outputEntries) != len(inputEntries) {
+				td.Fatalf(t, "parse results in %d entries, expected %d. Input:\n%s", len(outputEntries), len(inputEntries), entryStr)
+			}
+			// Show what has been decoded.
+			buf.WriteString("# after parse:\n")
+			for _, entry := range outputEntries {
+				fmt.Fprintf(&buf, "%# v\n", pretty.Formatter(entry))
+				if entry.StructuredEnd != 0 {
+					var payload interface{}
+					if err := json.Unmarshal([]byte(entry.Message[entry.StructuredStart:entry.StructuredEnd]), &payload); err != nil {
+						td.Fatalf(t, "JSON entry not parsable: %v", err)
+					}
+					fmt.Fprintf(&buf, "JSON payload in previous entry: %+v\n", payload)
+				}
+			}
+			return buf.String()
+
+		default:
+			t.Fatalf("unsupported command: %q", td.Cmd)
+		}
+		return ""
+		// unreachable
+	})
+}
+
+func TestCrdbV1EntryDecoderForVeryLargeEntries(t *testing.T) {
+	entryIdx := 1
+	formatEntry := func(s Severity, c Channel, now time.Time, gid int, file string, line int, tags, msg string) string {
+		entry := logpb.Entry{
+			Severity:  s,
+			Channel:   c,
+			Time:      now.UnixNano(),
+			Goroutine: int64(gid),
+			File:      file,
+			Line:      int64(line),
+			Tags:      tags,
+			Message:   msg,
+			Counter:   uint64(entryIdx),
+		}
+		entryIdx++
+		var buf bytes.Buffer
+		_ = FormatLegacyEntry(entry, &buf)
+		return buf.String()
+	}
+
+	t1 := timeutil.Now().Round(time.Microsecond)
+	t2 := t1.Add(time.Microsecond)
+
+	// Verify the truncation logic for reading logs that are longer than the
+	// default scanner can handle.
+	preambleLength := len(formatEntry(severity.INFO, channel.DEV, t1, 0, "clog_test.go", 136, ``, ""))
```

Comment:
Each instance of the string "clog_test.go" in `format_crdb_v1_test.go` and `testdata/crdb_v1` should be changed to the string "format_crdb_v1_test.go"